mapflow/mapflow.py

```diff
@@ -1968,17 +1981,38 @@ def create_processing_request(self,
         use_image_extent_as_aoi = self.dlg.useImageExtentAsAoi.isChecked()
         image_id = self.dlg.imageId.text()
         provider_name = None
-        selected_image = self.dlg.metadataTable.selectedItems()
-        if selected_image:
+        requires_id = False
+        selected_images = self.dlg.metadataTable.selectedItems()
+        if selected_images:
             try:
-                row = selected_image[0].row()
-                local_image_index = int(self.dlg.metadataTable.item(row, self.config.LOCAL_INDEX_COLUMN).text())
-                provider_name = self.search_footprints[local_image_index].attribute("providerName")
+                rows = list(set(image.row() for image in selected_images))
+                local_image_indices = [int(self.dlg.metadataTable.item(row, self.config.LOCAL_INDEX_COLUMN).text()) 
+                                       for row in rows]
+                provider_names = [self.search_footprints[local_image_index].attribute("providerName")
```

Comment:
I would move this logic of checking/selecting providers to some separate function, it looks too heavy here